Models/PriceVariantsPart.cs

```diff
@@ -0,0 +1,16 @@
+using System.Collections.Generic;
+using Money;
+using OrchardCore.ContentManagement;
+
+namespace OrchardCore.Commerce.Models
+{
+    /// <summary>
+    /// A product variants prices based on predefined attributes.
+    /// </summary>
+    public class PriceVariantsPart : ContentPart
+    {
+        public Amount BasePrice { get; set; }
```

Comment:
People must use only one `PricePart` or `PriceVariantsPart`. Otherwise we `PriceVariantsPart` will depend on `PricePart`. Let me know you thoughts.

Reply:
I think this is problematic.

One scenario is a business that starts with simple price. Then they add attributes to their existing products, that they want to individually price. If you have to choose between those two parts, you have no migration path, and you'll have to enter all those prices again.

Another scenario is one where you don't have a base price at all, only mandatory attributes with each a different price.

For at least those two reasons, but also for separation of concerns, I think the base price should be removed (and clearly, prices per attribute can't depend on the simple price part).

Reply:
All clear. So let me know best way to save variants:
1. Currency + Dictionary<string, decimal> properties.
2. Dictionary<string, Amount> property.

As for me first one is better b/c all variants must have same currency.

Reply:
I'm not sure why all variants should have the same currency actually. I could see a scenario where variants are geographically based, like a bookseller who sells translations of the same item. Just to be safe, I'd go for #2 because it offers that flexibility at a low cost (unless I'm missing something).

Reply:
Agree! Will do.